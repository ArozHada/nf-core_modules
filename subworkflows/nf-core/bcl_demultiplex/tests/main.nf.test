nextflow_workflow {

    name "Test Workflow BCL_DEMULTIPLEX"
    script "../main.nf"
    workflow "BCL_DEMULTIPLEX"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "bcl_demultiplex"
    tag "subworkflows/bcl_demultiplex"
    tag "bclconvert"
    tag "bcl2fastq"


    test("homo_sapiens - bclconvert") {

        when {
            workflow {
                """
                input[0] = Channel.value([
                            [id:'test', lane:1 ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_flowcell_samplesheet'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_flowcell'], checkIfExists: true)
                            ])

                input[1] = ['bclconvert']
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out.fastq).match("fastq") },
                { assert snapshot(workflow.out.interop).match("interop") },
                { assert snapshot(workflow.out.reports).match("reports") },
                { assert snapshot(workflow.out.versions).match("versions") },
            )
        }
    }

    test("homo_sapiens - bcl2fastq") {

        when {
            workflow {
                """
                input[0] = Channel.value([
                            [id:'test', lane:1 ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_flowcell_samplesheet'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_flowcell'], checkIfExists: true)
                            ])

                input[1] = ['bcl2fastq']
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out.fastq).match("fastq") },
                { assert snapshot(workflow.out.interop).match("interop") },
                { assert snapshot(workflow.out.reports).match("reports") },
                { assert snapshot(workflow.out.versions).match("versions") },
            )
        }
    }
}
