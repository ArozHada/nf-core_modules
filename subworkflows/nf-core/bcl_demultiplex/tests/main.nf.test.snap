{
    "reports_bclconvert": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    [
                        "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                        "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                        "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                        "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                        "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                        "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                        "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                        "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                        "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                        "SampleSheet.csv:md5,dc0dffd39541dd6cc5b4801d768a8d2b",
                        "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                        "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-29T17:14:56.60141244"
    },
    "versions_bclconvert": {
        "content": [
            [
                "versions.yml:md5,16bd2c2449c65bde2e76b42817dc5e0f"
            ]
        ],
        "timestamp": "2023-11-29T17:14:56.873984713"
    },
    "interop_bcl2fastq": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    [
                        "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
                        "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
                        "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
                        "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
                        "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                        "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
                        "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-29T17:15:12.303326257"
    },
    "fastq_bclconvert": {
        "content": [
            [
                [
                    {
                        "id": "Sample1_S1_L001",
                        "samplename": "Sample1",
                        "readgroup": {
                            "ID": "000000000-K9H97.1",
                            "PU": "000000000-K9H97.1",
                            "PL": "ILLUMINA",
                            "SM": "Sample1"
                        },
                        "fcid": "test",
                        "lane": 1,
                        "single_end": true
                    },
                    [
                        "Sample1_S1_L001_R1_001.fastq.gz:md5,a0c1359a2c922967660524a6e77db494"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-29T17:14:56.050165025"
    },
    "homo_sapiens - bcl2fastq": {
        "content": [
            [
                "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
                "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
                "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
                "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
                "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
                "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
            ]
        ],
        "timestamp": "2023-11-29T17:15:12.813335831"
    },
    "interop_bclconvert": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    [
                        "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
                        "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
                        "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
                        "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
                        "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
                        "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-29T17:14:56.334419526"
    },
    "reports_bcl2fastq": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    [
                        [
                            [
                                [
                                    [
                                        [
                                            "lane.html:md5,794e48287f47a9f22dcb6b6d0c22c3eb",
                                            "laneBarcode.html:md5,2bbdae3ee57151eab520c966597d7438"
                                        ],
                                        [

                                        ]
                                    ]
                                ],
                                [
                                    [
                                        [
                                            "lane.html:md5,f741870307050dcea79a838f5971770f",
                                            "laneBarcode.html:md5,ffe2e863811c76cb9da27d5d124e0611"
                                        ],
                                        [
                                            "lane.html:md5,cca97e771d3491dbc8cd3fe389595b09",
                                            "laneBarcode.html:md5,cca97e771d3491dbc8cd3fe389595b09"
                                        ]
                                    ],
                                    [
                                        [
                                            "lane.html:md5,c383b0768d9978733d3f5d3b91c15af0",
                                            "laneBarcode.html:md5,48842c23b9a2816aec540177df870968"
                                        ],
                                        [

                                        ]
                                    ]
                                ]
                            ],
                            "Report.css:md5,eb7d3eb68fc1539f411404987246b59b",
                            "index.html:md5,5747c407854ae2c358d0ec201ce622d8",
                            "tree.html:md5,a1b9bf592973ca829ec69ddf888b7e34"
                        ]
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-29T17:15:13.337907541"
    },
    "fastq_bcl2fastq": {
        "content": [
            [
                [
                    {
                        "id": "Sample1_S1_L001",
                        "samplename": "Sample1",
                        "readgroup": {
                            "ID": "000000000-K9H97.1",
                            "PU": "000000000-K9H97.1",
                            "PL": "ILLUMINA",
                            "SM": "Sample1"
                        },
                        "fcid": "test",
                        "lane": 1,
                        "single_end": true
                    },
                    [
                        "Sample1_S1_L001_R1_001.fastq.gz:md5,e92fce7b86c6447b053d72c5cb4be89c"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-11-29T17:15:11.806792352"
    },
    "versions_bcl2fastq": {
        "content": [
            [
                "versions.yml:md5,513271615cb02dbe541a6202d713ef81"
            ]
        ],
        "timestamp": "2023-11-29T17:15:13.840743366"
    }
}
